Cairo Desktop/Cairo Desktop/Startup.cs

```diff
@@ -84,13 +86,60 @@ public static void Main(string[] args)
             app.InitializeComponent();  // This sets up the Unhandled Exception stuff... 
 
             // Themes are very UI centric. We should devise a way of having Plugins/Extensions contribute to this.
-            string theme = Settings.CairoTheme;
+            string theme = Settings.Instance.CairoTheme;
             if (theme != "Default")
-                if (System.IO.File.Exists(AppDomain.CurrentDomain.BaseDirectory + theme))
-                    app.Resources.MergedDictionaries.Add((ResourceDictionary)XamlReader.Load(System.Xml.XmlReader.Create(AppDomain.CurrentDomain.BaseDirectory + theme)));
+            {
+                string themeFilePath = AppDomain.CurrentDomain.BaseDirectory + theme;
+                if (System.IO.File.Exists(themeFilePath))
+                {
+                    ResourceDictionary newRes = new ResourceDictionary();
+                    newRes.Source = new Uri(themeFilePath, UriKind.RelativeOrAbsolute);
+                    app.Resources.MergedDictionaries.Add(newRes);
+                }
+            }
+
+            Settings.Instance.PropertyChanged += (s, e) =>
+              {
+                  if (e != null && !string.IsNullOrWhiteSpace(e.PropertyName) && e.PropertyName == "CairoTheme")
+                  {
+                      App.Current.Resources.MergedDictionaries.Clear();
+                      ResourceDictionary cairoResource = new ResourceDictionary();
+
+                      // Put our base theme back
+                      cairoResource.Source = new Uri("Cairo.xaml", UriKind.RelativeOrAbsolute);
+                      App.Current.Resources.MergedDictionaries.Add(cairoResource);
+
+                      string newTheme = Settings.Instance.CairoTheme;
+                      if (newTheme != "Default")
+                      {
+                          string newThemeFilePath = AppDomain.CurrentDomain.BaseDirectory + newTheme;
+                          if (System.IO.File.Exists(newThemeFilePath))
+                          {
+                              ResourceDictionary newRes = new ResourceDictionary();
+                              newRes.Source = new Uri(newThemeFilePath, UriKind.RelativeOrAbsolute);
+                              app.Resources.MergedDictionaries.Add(newRes);
+                          }
+                      }
+
+                      // Update pesky Image resources
```

Comment:
Any idea why this is happening to only these icons? Are other icons switching like they should?

Reply:
Not sure. those are the ones i saw. I think that the imagesources must not be getting notified that things have changed